src/streamsync/core.py

```diff
@@ -1125,10 +1250,68 @@ def handle(self, ev: StreamsyncEvent) -> StreamsyncEventResult:
         return {"ok": ok, "result": result}
 
 
-state_serialiser = StateSerialiser()
-initial_state = StreamsyncState()
-base_component_tree = ComponentTree()
-session_manager = SessionManager()
+class DictPropertyProxy:
+    """
+    A descriptor based recipe that makes it possible to write shorthands
+    that forward attribute access from one object onto another.
+
+    >>> class A:
+    >>>     foo: int = DictPropertyProxy("proxy_state", "prop1")
+    >>>     bar: int = DictPropertyProxy("proxy_state", "prop2")
+    >>>
+    >>>     def __init__(self):
+    >>>         self._state_proxy = StateProxy({"prop1": 1, "prop2": 2})
+    >>>
+    >>> a = A()
+    >>> print(a.foo)
+
+    This descriptor avoids writing the code below to establish a proxy
+     with a child instance
+
+    >>> class A:
+    >>>
+    >>>     def __init__(self):
+    >>>         self._state_proxy = StateProxy({"prop1": 1, "prop2": 2})
+    >>>
+    >>>     @property
+    >>>     def prop1(self):
+    >>>         return self._state_proxy['prop1']
+    >>>
+    >>>     @foo.setter
+    >>>     def prop1(self, value):
+    >>>         self._state_proxy['prop1'] = value
+    >>>
+    """
+
+    def __init__(self, objectName, key):
+        self.objectName = objectName
+        self.key = key
+
+    def __get__(self, instance, owner=None):
+        proxy = getattr(instance, self.objectName)
+        return proxy[self.key]
+
+    def __set__(self, instance, value):
+        proxy = getattr(instance, self.objectName)
+        proxy[self.key] = value
+
+S = TypeVar("S", bound=StreamsyncState)
+
+def new_initial_state(klass: Type[S]) -> S:
+    """
+    Initializes the initial state of the application and makes it globally accessible.
+
+    The class used for the initial state must be a subclass of StreamsyncState.
+
+    >>> class MyState(StreamsyncState):
+    >>>     pass
+    >>>
+    >>> initial_state = new_initial_state(MyState)
+    """
+    global initial_state
```

Comment:
Rewriting a module variable there introduces a risk of introducing futur bugs with current accessors. When a variable is import from external module, its value is freezed. There is a risk to replace the value of the variable without updating external module.

To reduce the risk of issue, we should expose a function that allow to get the variable `initial_state()`. Using a function avoid the risk for this issue.

Reply:
So, keeping the global variable but using an accessor rather than importing it directly?

Like this?

```py
def get_initial_state():
  return initial_state
```

Reply:
yes exactly